nilearn/regions/connectivity.py

```diff
@@ -0,0 +1,392 @@
+        idx1 = idx[0] * prod(sz[1:3]) + idx[1] * sz[2] + idx[2]
+    else:
+        idx1 = idx[:, 0] * prod(sz[1:3]) + idx[:, 1] * sz[2] + idx[:, 2]
+    return idx1
+
+
+def make_local_connectivity_scorr(func_img, clust_mask_img, thresh):
+    """
+    Constructs a spatially constrained connectivity matrix from a fMRI dataset.
+    The weights w_ij of the connectivity matrix W correspond to the
+    spatial correlation between the whole brain FC maps generated from the
+    time series from voxel i and voxel j. Connectivity is only calculated
+    between a voxel and the 27 voxels in its 3D neighborhood
+    (face touching and edge touching).
+
+    Parameters
+    ----------
+    func_img : Nifti1Image
+        4D Nifti1Image containing fMRI data.
+    clust_mask_img : Nifti1Image
+        3D NIFTI file containing a mask, which restricts the voxels used in
+        the analysis.
+    thresh : str
+        Threshold value, correlation coefficients lower than this value
+        will be removed from the matrix (set to zero).
+
+    Returns
+    -------
+    W : Compressed Sparse Matrix
+        A Scipy sparse matrix, with weights corresponding to the spatial
+        correlation between the time series from voxel i and voxel j
+
+    References
+    ----------
+    .. [1] Craddock, R. C., James, G. A., Holtzheimer, P. E., Hu, X. P., &
+      Mayberg, H. S. (2012). A whole brain fMRI atlas generated via
+      spatially constrained spectral clustering. Human Brain Mapping.
+      https://doi.org/10.1002/hbm.21333
+
+    """
+    from scipy.sparse import csc_matrix
+    from itertools import product
+
+    neighbors = np.array(
+        sorted(
+            sorted(
+                sorted(
+                    [list(x) for x in list(set(product({-1, 0, 1},
+                                                       repeat=3)))],
+                    key=lambda k: (k[0]),
+                ),
+                key=lambda k: (k[1]),
+            ),
+            key=lambda k: (k[2]),
+        )
+    )
+
+    # Read in the mask
+    msz = clust_mask_img.shape
+
+    # Convert the 3D mask array into a 1D vector
+    mskdat = np.reshape(
+        np.asarray(
+            clust_mask_img.dataobj).astype("bool"),
+        prod(msz))
+
+    # Determine the 1D coordinates of the non-zero elements of the mask
+    iv = np.nonzero(mskdat)[0]
+    sz = func_img.shape
+
+    # Reshape fmri data to a num_voxels x num_timepoints array
+    func_data = _safe_get_data(func_img).astype(np.float32)
+    func_data = np.reshape(func_data, (prod(sz[:3]), sz[3]))
+
+    # Mask the datset to only the in-mask voxels
+    func_data = func_data[iv, :]
+    func_data_sz = func_data.shape
+
+    # Z-score fmri time courses, this makes calculation of the
+    # correlation coefficient a simple matrix product
+    func_data_s = np.tile(np.std(func_data, 1), (func_data_sz[1], 1)).T
+
+    # Replace 0 with large number to avoid div by zero
+    func_data_s[func_data_s == 0] = 1000000
```

Comment:
you don't need a large number...

Reply:
I could also just incorporate the out=np.zeros_like(a) and where=b!=0 args in np.divide everywhere there's a division operation if that'd be preferable?